tests/Unit/MessengerContextTest.php

```diff
@@ -0,0 +1,640 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BehatMessengerContext\Tests\Unit;
+
+use Behat\Gherkin\Node\PyStringNode;
+use BehatMessengerContext\Context\MessengerContext;
+use BehatMessengerContext\Context\TransportRetriever;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\InMemory\InMemoryTransport;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+class MessengerContextTest extends TestCase
+{
+    private MessengerContext $messengerContext;
+
+    /** @var ContainerInterface&MockObject */
+    private ContainerInterface $container;
+
+    /** @var NormalizerInterface&MockObject */
+    private NormalizerInterface $normalizer;
+
+    /** @var TransportRetriever&MockObject */
+    private TransportRetriever $transportRetriever;
+
+    /** @var InMemoryTransport&MockObject */
+    private InMemoryTransport $inMemoryTransport;
+
+    protected function setUp(): void
+    {
+        $this->container = $this->createMock(Container::class);
+        $this->normalizer = $this->createMock(NormalizerInterface::class);
+        $this->transportRetriever = $this->createMock(TransportRetriever::class);
+        $this->inMemoryTransport = $this->createMock(InMemoryTransport::class);
+
+        $this->messengerContext = new MessengerContext(
+            $this->container,
+            $this->normalizer,
+            $this->transportRetriever
+        );
+    }
+
+    public function testTransportShouldContainMessageWithJson(): void
+    {
+        $message = new \stdClass();
+        $expectedMessage = ['key' => 'value'];
+
+        $this->inMemoryTransport
+            ->expects($this->once())
+            ->method('get')
+            ->willReturn([new \Symfony\Component\Messenger\Envelope($message)]);
```

Comment:
Here and below too pleasse
```suggestion
            ->willReturn([new Envelope($message)]);
```